codersdk/workspacebuilds.go

```diff
@@ -186,17 +186,28 @@ type ProvisionerTiming struct {
 }
 
 type AgentScriptTiming struct {
-	StartedAt   time.Time `json:"started_at" format:"date-time"`
-	EndedAt     time.Time `json:"ended_at" format:"date-time"`
-	ExitCode    int32     `json:"exit_code"`
-	Stage       string    `json:"stage"`
-	Status      string    `json:"status"`
-	DisplayName string    `json:"display_name"`
+	StartedAt          time.Time `json:"started_at" format:"date-time"`
+	EndedAt            time.Time `json:"ended_at" format:"date-time"`
+	ExitCode           int32     `json:"exit_code"`
+	Stage              string    `json:"stage"`
+	Status             string    `json:"status"`
+	DisplayName        string    `json:"display_name"`
+	WorkspaceAgentID   string    `json:"workspace_agent_id"`
+	WorkspaceAgentName string    `json:"workspace_agent_name"`
+}
+
+type AgentConnectionTiming struct {
+	StartedAt          time.Time `json:"started_at" format:"date-time"`
+	EndedAt            time.Time `json:"ended_at" format:"date-time"`
+	Stage              string    `json:"stage"`
+	WorkspaceAgentID   string    `json:"workspace_agent_id"`
+	WorkspaceAgentName string    `json:"workspace_agent_name"`
 }
 
 type WorkspaceBuildTimings struct {
-	ProvisionerTimings []ProvisionerTiming `json:"provisioner_timings"`
-	AgentScriptTimings []AgentScriptTiming `json:"agent_script_timings"`
+	ProvisionerTimings     []ProvisionerTiming     `json:"provisioner_timings"`
+	AgentScriptTimings     []AgentScriptTiming     `json:"agent_script_timings"`
+	AgentConnectionTimings []AgentConnectionTiming `json:"agent_connection_timings"`
```

Comment:
It feels weird to me that this is a separate member on this struct.
Logically, all timings should be grouped by the agent. So the script & connection timings should both be on the same `AgentTimings` struct (for example).

Reply:
I wondered about the same issue but didn't want to introduce a breaking change. However, since the agent script timings have not been released yet, it shouldn't be a problem.

Reply:
Wdyt about this structure?

```golang
type AgentScriptTiming struct {
	StartedAt          time.Time `json:"started_at" format:"date-time"`
	EndedAt            time.Time `json:"ended_at" format:"date-time"`
	ExitCode           int32     `json:"exit_code"`
	Stage              string    `json:"stage"`
	Status             string    `json:"status"`
	DisplayName        string    `json:"display_name"`
	WorkspaceAgentID   string    `json:"workspace_agent_id"`
	WorkspaceAgentName string    `json:"workspace_agent_name"`
}

type AgentConnectionTiming struct {
	StartedAt          time.Time `json:"started_at" format:"date-time"`
	EndedAt            time.Time `json:"ended_at" format:"date-time"`
	Stage              string    `json:"stage"`
	WorkspaceAgentID   string    `json:"workspace_agent_id"`
	WorkspaceAgentName string    `json:"workspace_agent_name"`
}

type AgentTiming type struct {
	AgentID string
	AgentName string
	Scripts    []AgentScriptTiming
	Connection AgentConnectionTiming
}

type WorkspaceBuildTimings struct {
	ProvisionerTimings []ProvisionerTiming `json:"provisioner_timings"`
	AgentTimings       []AgentTiming       `json:"agent_timings"`
}
```

Reply:
> I wondered about the same issue but didn't want to introduce a breaking change. However, since the agent script timings have not been released yet, it shouldn't be a problem.

I'm afraid that's incorrect. This feature is already in our mainline release: https://coder.com/docs/reference/api/workspaces#get-workspace-timings-by-id

We'll have to keep the current structure.